src/map/attack.cpp

```diff
@@ -99,16 +99,40 @@ void CAttack::SetCritical(bool value)
     }
     else
     {
-        float attBonus = 0.f;
+        float attBonus = 1.0f;
         if (m_attackType == PHYSICAL_ATTACK_TYPE::KICK)
         {
             if (CStatusEffect* footworkEffect = m_attacker->StatusEffectContainer->GetStatusEffect(EFFECT_FOOTWORK))
             {
-                attBonus = footworkEffect->GetSubPower() / 256.f; // Mod is out of 256
+                attBonus = 1 + (footworkEffect->GetSubPower() / 256.f); // Mod is out of 256
             }
         }
 
-        m_damageRatio = battleutils::GetDamageRatio(m_attacker, m_victim, m_isCritical, attBonus);
+        SKILLTYPE skilltype = SKILLTYPE::SKILL_NONE;
```

Comment:
@WinterSolstice8 I am a bit confused about this block of code. Why not use the `GetWeaponSlot` function of attack.cpp and also `GetWeaponSlot` has the logic `return m_attackDirection == RIGHTATTACK ? SLOT_MAIN : SLOT_SUB;` but this code seems to have the reverse logic? Also is there need for two `m_attacker->objtype == TYPE_PC` checks? I might be missing something.

Reply:
"right attack" is the "right hand" animation, aka main hand, aka SLOT_MAIN

Reply:
as for the other issue that was probably just a mistake

Reply:
I don't really feel like looking more at code right now, but I see what you're saying

Reply:
Looks like it should be this, but can't really test well in my current rebase state. Will check tomorrow probably

```
        SKILLTYPE skilltype = SKILLTYPE::SKILL_NONE;

        if (m_attacker->objtype == TYPE_PC)
        {
            SLOTTYPE slot = static_cast<SLOTTYPE>(GetWeaponSlot());
            
            if (auto* weapon = dynamic_cast<CItemWeapon*>(m_attacker->m_Weapons[slot]))
            {
                skilltype = static_cast<SKILLTYPE>(weapon->getSkillType());
            }
            else // barehanded
            {
                skilltype = SKILLTYPE::SKILL_HAND_TO_HAND;
            }
        }
```

Reply:
> Looks like it should be this, but can't really test well in my current rebase state. Will check tomorrow probably
> 
> ```
>         SKILLTYPE skilltype = SKILLTYPE::SKILL_NONE;
> 
>         if (m_attacker->objtype == TYPE_PC)
>         {
>             SLOTTYPE slot = static_cast<SLOTTYPE>(GetWeaponSlot());
>             
>             if (auto* weapon = dynamic_cast<CItemWeapon*>(m_attacker->m_Weapons[slot]))
>             {
>                 skilltype = static_cast<SKILLTYPE>(weapon->getSkillType());
>             }
>             else // barehanded
>             {
>                 skilltype = SKILLTYPE::SKILL_HAND_TO_HAND;
>             }
>         }
> ```

Yep, as I need the weaponslot for the call later of `GetDamageRatio` just below this, I moved it up a bit and made a [PR](https://github.com/LandSandBoat/server/pull/5459) with your fix (as I was working in the area in any case with ASB stuff) and added you as co-author. @WinterSolstice8 Thanks!